arangod/RocksDBEngine/RocksDBEngine.cpp

```diff
@@ -2465,6 +2473,72 @@ RocksDBFilePurgeEnabler RocksDBEngine::startPurging() noexcept {
   return RocksDBFilePurgeEnabler(this);
 }
 
+void RocksDBEngine::trackLowerBoundToKeep(uint64_t tick) {
+  TRI_ASSERT(tick != 0);
+
+  WRITE_LOCKER(lock, _walFileLock);
+
+  auto it = _lowerBoundsToKeep.find(tick);
+  if (it == _lowerBoundsToKeep.end()) {
+    _lowerBoundsToKeep.emplace(tick, 1);
+  } else {
+    // increase counter
+    ++(it->second);
+  }
```

Comment:
NIT: could be simplified to this:
```suggestion
  // increase counter
  ++_lowerBoundsToKeep[tick];
```